src/codegen/mod.rs

```diff
@@ -889,6 +891,11 @@ impl CodeGenerator for CompInfo {
         if item.can_derive_debug(ctx, ()) {
             derives.push("Debug");
         }
+        if item.can_derive_default(ctx, ()) {
+            derives.push("Default");
+        } else {
+            needs_default_impl = true;
```

Comment:
This needs to be `needs_default_impl = ctx.options().derive_default`. Please add a test for this (see `tests/headers/no-derive-debug.h` for a similar example).